src/main/java/net/minecraftforge/event/entity/player/TradeWithVillagerEvent.java

```diff
@@ -0,0 +1,46 @@
+package net.minecraftforge.event.entity.player;
+
+import net.minecraft.world.entity.npc.AbstractVillager;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.trading.MerchantOffer;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.eventbus.api.Cancelable;
+
+/**
+ * TradeWithVillagerEvent is fired when a player trades with
+ * a villager inheriting from {@link AbstractVillager}. <br>
+ * <br>
+ * This event is not {@link Cancelable}.<br>
+ * <br>
+ * This event does not have a result. {@link HasResult} <br>
+ * <br>
+ * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
```

Comment:
Here's a suggestion for improved javadocs that also follow the formatting I made for the client-side event javadocs. 

```suggestion
 * Fired when a player trades with an {@link AbstractVillager}.
 *
 * <p>This event is not {@linkplain Cancelable cancellable}, and does not {@linkplain Event.HasResult have a result}.</p>
 *
 * <p>This event is fired on the {@linkplain MinecraftForge#EVENT_BUS main Forge event bus},
 * only on the {@linkplain LogicalSide#SERVER logical server}.</p>
```